### ISSUE TITLE
fix: block .local mDNS hostnames in MCP server SSRF validation

### ISSUE THREAD

**Issue by purupurupu** [purupurupu/css-noop-checker#172]
## Summary
`url-validation.ts` in the MCP server doesn't block `.local` mDNS hostnames. `http://printer.local/` or `http://router.local/` would pass validation and be opened by Playwright, resolving to local network devices via mDNS.

## Details
- `mcp-server/src/url-validation.ts:20-51` — `isPrivateHost()` checks localhost, private IPs, but not internal TLDs

## Fix
Add suffix checks in `isPrivateHost()`:
```ts
if (lower.endsWith('.local') || lower.endsWith('.local.')) return true;
if (lower.endsWith('.localhost')) return true;
if (lower.endsWith('.internal')) return true;
if (lower.endsWith('.home.arpa')) return true;
```

Also consider blocking `.onion` (Tor hidden services).
